apiserver/facades/client/machinemanager/state.go

```diff
@@ -44,47 +59,20 @@ func (s stateShim) Machine(name string) (Machine, error) {
 	return machineShim{m}, nil
 }
 
-func (s stateShim) ModelConfig() (*config.Config, error) {
-	return s.State.ModelConfig()
-}
-
 func (s stateShim) Model() (Model, error) {
 	return s.State.Model()
 }
-func (s stateShim) ModelTag() names.ModelTag {
-	return s.State.ModelTag()
-}
 
-func (s stateShim) GetBlockForType(t state.BlockType) (state.Block, bool, error) {
-	return s.State.GetBlockForType(t)
+type poolShim struct {
+	pool *state.StatePool
 }
 
-func (s stateShim) AddOneMachine(template state.MachineTemplate) (*state.Machine, error) {
-	return s.State.AddOneMachine(template)
-}
-
-func (s stateShim) AddMachineInsideNewMachine(template, parentTemplate state.MachineTemplate, containerType instance.ContainerType) (*state.Machine, error) {
-	return s.State.AddMachineInsideNewMachine(template, parentTemplate, containerType)
-}
-
-func (s stateShim) AddMachineInsideMachine(template state.MachineTemplate, parentId string, containerType instance.ContainerType) (*state.Machine, error) {
-	return s.State.AddMachineInsideMachine(template, parentId, containerType)
-}
-
-type Model interface {
-	Name() string
-	UUID() string
-	Cloud() string
-	CloudCredential() (names.CloudCredentialTag, bool)
-	CloudRegion() string
-	ModelTag() names.ModelTag
-	Config() (*config.Config, error)
-}
-
-type Machine interface {
-	Destroy() error
-	ForceDestroy() error
-	Units() ([]Unit, error)
+func (p *poolShim) GetModel(uuid string) (Model, func(), error) {
+	m, release, err := p.pool.GetModel(uuid)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	return m, func() { release() }, nil
```

Comment:
Nope. The release func returned by GetModel is `func () bool` but the Pool interface simplifies that to `func()`